src/Microsoft.IdentityModel.Tokens/Experimental/Validation/Results/Details/ValidationError.cs

```diff
@@ -131,44 +133,155 @@ public ValidationError AddStackFrame(StackFrame stackFrame)
             return this;
         }
 
+        /// <summary>
+        /// Adds a <see cref="StackFrame"/> to the list of stack frames and returns the updated object.
+        /// </summary>
+        /// <param name="stackFrame">
+        /// The <see cref="StackFrame"/> to use as a template if a cached frame does not exist.
+        /// If <c>null</c>, a new <see cref="StackFrame"/> is created.
+        /// </param>
+        /// <param name="memberName">
+        /// The name of the calling member. Automatically provided by the compiler.
+        /// Used as part of the cache key for the stack frame.
+        /// </param>
+        /// <param name="filePath">
+        /// The full path of the source file containing the caller. Automatically provided by the compiler.
+        /// Used as part of the cache key for the stack frame.
+        /// </param>
+        /// <param name="lineNumber">
+        /// The line number in the source file at which this method is called. Automatically provided by the compiler.
+        /// Used as part of the cache key for the stack frame.
+        /// </param>
+        /// <returns>
+        /// The updated <see cref="ValidationError"/> instance with the new <see cref="StackFrame"/> added to <see cref="StackFrames"/>.
+        /// </returns>
+        public ValidationError AddAsyncStackFrame(
+            StackFrame stackFrame,
+            [CallerMemberName] string memberName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0)
+        {
+            string key = memberName + filePath + lineNumber;
+            StackFrame cachedFrame = CachedStackFrames.GetOrAdd(key, _ =>
+            {
+                StackFrame frame = stackFrame ?? new StackFrame(1, true);
```

Comment:
When passing `1` to skip a frame, which frame is skipped, is it the GetOrAdd lambda?